ppa/archive/management/commands/hathi_rsync.py

```diff
@@ -40,52 +42,88 @@ def handle(self, *args, **kwargs):
         # in the database and not suppressed
         if htids:
             digworks = digworks.filter(source_id__in=htids)
-        # NOTE: report here on any skipped ids?
 
         # generate a list of unique source ids from the queryset
-        hathi_ids = digworks.values_list("source_id", flat=True).distinct()
-        self.stdout.write("Synchronizing data for %d records" % len(hathi_ids))
+        working_htids = digworks.values_list("source_id", flat=True).distinct()
+
+        # if htids were explicitly specified, report if any are skipped
+        if htids:
+            skipped_htids = set(htids) - set(working_htids)
+            if skipped_htids:
+                self.stdout.write(
+                    self.style.NOTICE(
+                        "Some ids not found in public HathiTrust volumes; skipping %s"
+                        % " ".join(skipped_htids)
+                    )
+                )
+
+        # bail out if there's nothing to do
+        # (e.g., explicit htids only and none valid)
+        if not working_htids:
+            return
```

Comment:
It seems worth logging/reporting that there's nothing to synchronize before exiting

Reply:
Ah, good catch. In earlier versions it output "synchronizing data for 0 records" but I've changed the order.